### ISSUE TITLE
Records do not line up with data model. The field 'first' is in data_model but not in a record

### ISSUE THREAD

**Issue by kevalhb** [dedupeio/dedupe#738]
I am using a cursor that's connecting to sql server. While I am trying to create a sample for my dedupe object, it throws me an error written in the object. I am trying to follow the mysql example. Unable to understand what's the issue.

Cross-checked database connection and temp_d variable values. Looks fine:

Here's my code snippet:
try:
    dbconnect = pypyodbc.connect('Driver={SQL Server};Server=servername;Database=dbname;')
    print("Database connection established")
except:
    print("Failed Connection")

fields = [{'field' : 'first' ,'variable name' : 'first' ,'type' : 'String', 'has missing' : True },
          {'field' : 'last', 'variable name' : 'last' ,'type' : 'String' , 'has missing' : True },
          {'field' : 'age','variable name' : 'age' ,'type' : 'ShortString' },
          {'field' : 'state','variable name' : 'state' ,'type' : 'ShortString', 'has missing' : True },
          {'field' : 'email','variable name' : 'email' ,'type' : 'String' },
          {'field' : 'Organization','variable name' : 'organization' ,'type' : 'String', 'has missing' : True}]

deduper = dedupe.Dedupe(fields, num_cores=3)

with dbconnect.cursor() as c_deduper:
  res =  c_deduper.execute("SELECT first,last,age,state,email,Organization From dbo.sampledata")
  temp_d = dict((i, row) for i, row in enumerate(c_deduper))
  deduper.sample(temp_d,1000)
  del temp_d

![model record mismatch error](https://user-images.githubusercontent.com/32277001/54953900-b45bb400-4f1f-11e9-8dbe-381b47e656f5.JPG)
![temp_d dictonary variable values](https://user-images.githubusercontent.com/32277001/54953902-b45bb400-4f1f-11e9-831c-7555d990f77a.JPG)

I am using Python 3.6 Not sure where I am going wrong. Is my dictionary created correctly?

**Comment by ChadMGeek** [comment]
I am working on a similar project and using the MySQL example as a template even though I'm working in SQL. I ran into the same error. So, what the code you are using is doing is creating a dictionary where the key is the row number and the value is a comma separated list of the values in the row. What the "sample" function is looking for is a dictionary of dictionaries, where the outer dictionary has the row number as the key and another dictionary as the value. The inner dictionary should have the column name as the key and the value as the value. Here is how I solved it:

`    c.execute(PERSON_SELECT)
    columns = [column[0] for column in c.description
    temp_d = dict((i, dict(zip(columns, row))) for i, row in enumerate(c))]`
